src/gardenlinux/github/__main__.py

```diff
@@ -0,0 +1,836 @@
+                    output += f"  download_url: {download_url}\n"
+
+                    if "regions" in data:
+                        output += "  regions:\n"
+                        for region in data["regions"]:
+                            if "image_name" in region:
+                                output += f"    - region: {region['region']}\n"
+                                output += f"      image_id: {region['image_id']}\n"
+                                output += f"      image_name: {region['image_name']}\n"
+                            else:
+                                output += f"    - region: {region['region']}\n"
+                                output += f"      image_id: {region['image_id']}\n"
+                    elif "details" in data and platform != "gcp":
+                        output += "  details:\n"
+                        for key, value in data["details"].items():
+                            output += f"    {key}: {value}\n"
+                    elif platform == "gcp" and "details" in data:
+                        # For GCP, move details up to same level as flavor
+                        for key, value in data["details"].items():
+                            output += f"  {key}: {value}\n"
+                    elif "gallery_images" in data or "marketplace_images" in data:
+                        if data.get("gallery_images"):
+                            output += "  gallery_images:\n"
+                            for img in data["gallery_images"]:
+                                output += f"    - hyper_v_generation: {img['hyper_v_generation']}\n"
+                                output += f"      azure_cloud: {img['azure_cloud']}\n"
+                                output += f"      image_id: {img['image_id']}\n"
+                        if data.get("marketplace_images"):
+                            output += "  marketplace_images:\n"
+                            for img in data["marketplace_images"]:
+                                output += f"    - hyper_v_generation: {img['hyper_v_generation']}\n"
+                                output += f"      urn: {img['urn']}\n"
+
+                output += "```\n\n"
+                output += "</details>\n\n"
+
+            output += "</details>\n\n"
+
+        output += "</details>\n\n"
+
+    return output
+
+
+def download_metadata_file(
+    s3_artifacts, cname, version, commitish_short, artifacts_dir
+):
+    """
+    Download metadata file (s3_metadata.yaml)
+    """
+    LOGGER.debug(
+        f"{s3_artifacts=} | {cname=} | {version=} | {commitish_short=} | {artifacts_dir=}"
+    )
+    release_object = list(
+        s3_artifacts._bucket.objects.filter(
+            Prefix=f"meta/singles/{cname}-{version}-{commitish_short}"
+        )
+    )[0]
+    LOGGER.debug(f"{release_object.bucket_name=} | {release_object.key=}")
+
+    s3_artifacts._bucket.download_file(
+        release_object.key, artifacts_dir.joinpath(f"{cname}.s3_metadata.yaml")
+    )
```

Comment:
`s3_artifacts` provides a property called `bucket` to not use the "private" property `_bucket`. If it could be a common use case to download the S3 metadata file (only / separately) please consider adding a method in `gardenlinux.s3.S3Artifacts` for it.

Reply:
Still would like to see `s3_artifacts.bucket` here and everywhere else where used. Reasoning given last time.